simd/src/Kokkos_SIMD_AVX2.hpp

```diff
@@ -563,6 +646,189 @@ simd<double, simd_abi::avx2_fixed_size<4>> condition(
                        static_cast<__m256d>(a)));
 }
 
+template <>
+class simd<float, simd_abi::avx2_fixed_size<4>> {
+  __m128 m_value;
+
+ public:
+  using value_type = float;
+  using abi_type   = simd_abi::avx2_fixed_size<4>;
+  using mask_type  = simd_mask<value_type, abi_type>;
+  using reference  = value_type&;
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION simd()            = default;
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION simd(simd const&) = default;
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION simd(simd&&)      = default;
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION simd& operator=(simd const&) = default;
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION simd& operator=(simd&&) = default;
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION static constexpr std::size_t size() {
+    return 4;
+  }
+  template <typename U, std::enable_if_t<std::is_convertible_v<U, value_type>,
+                                         bool> = false>
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION simd(U&& value)
+      : m_value(_mm_set1_ps(value_type(value))) {}
+  template <typename G,
+            std::enable_if_t<
+                std::is_invocable_r_v<value_type, G,
+                                      std::integral_constant<std::size_t, 0>>,
+                bool> = false>
+  KOKKOS_FORCEINLINE_FUNCTION simd(G&& gen)
+      : m_value(_mm_setr_ps(gen(std::integral_constant<std::size_t, 0>()),
+                            gen(std::integral_constant<std::size_t, 1>()),
+                            gen(std::integral_constant<std::size_t, 2>()),
+                            gen(std::integral_constant<std::size_t, 3>()))) {}
```

Comment:
It looks like the double precision specialization is missing that generator constructor.

Reply:
Generator constructor for double precision was added in #6347.
After a rebase, it should be there.